src/Data/Profunctor/Cayley.hs

```diff
@@ -26,6 +26,9 @@ import Prelude hiding ((.), id)
 -- static arrows
 newtype Cayley f p a b = Cayley { runCayley :: f (p a b) }
 
+instance (Functor f, Functor (p a)) => Functor (Cayley f p a) where
```

Comment:
This use of `FlexibleContexts` here is a bit curious. After all, `Cayley` is isomorphic to [`Tannen`](https://hackage.haskell.org/package/bifunctors-5.5.7/docs/Data-Bifunctor-Tannen.html#t:Tannen) from the `bifunctors` package, and `Tannen` [already has a `Functor` instance](https://hackage.haskell.org/package/bifunctors-5.5.7/docs/src/Data.Bifunctor.Tannen.html#line-159):

```hs
instance (Functor f, Bifunctor p) => Functor (Tannen f p a) where
  fmap f = Tannen . fmap (B.second f) . runTannen
```

This makes me wonder if a `Functor` instance for `Cayley` should use the same instance context. Alas, it cannot under the new version of `Profunctor` with a quantified superclass! If it does, this will cause a ripple effect on downstream instances, such as the `Profunctor` instance for `Cayley`:

```
src/Data/Profunctor/Cayley.hs:46:10: error:
    • Could not deduce (Bifunctor p)
        arising from the superclasses of an instance declaration
      from the context: (Functor f, Profunctor p)
        bound by the instance declaration
        at src/Data/Profunctor/Cayley.hs:46:10-61
      Possible fix:
        add (Bifunctor p) to the context of
          a quantified context
          or the instance declaration
    • In the instance declaration for ‘Profunctor (Cayley f p)’
   |
46 | instance (Functor f, Profunctor p) => Profunctor (Cayley f p) where
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Yikes! Having a `Bifunctor p` and `Profunctor p` constraint simultaneously is definitely fishy.

One possible remedy is to change the `Functor` instance for `Cayley` to use a flexible `Functor (p a)` constraint, much like what is being proposed here. This, however, defeats one of the implicit goals of the `bifunctors` package, which is to allow working with higher-kinded data types in a way that avoids `FlexibleContexts`. However, I can't see another way to do this that simultaneously allows for `Profunctor` to have a quantified superclass.

Reply:
A heavy-handed fix is to change the `Functor` instance for `Cayley` to use a quantified constraint:

```diff
-instance (Functor f, Bifunctor p) => Functor (Tannen f p a) where
-  fmap f = Tannen . fmap (B.second f) . runTannen
+instance (Functor f, forall x. Functor (p x)) => Functor (Tannen f p a) where
+  fmap f = Tannen . fmap (fmap f) . runTannen
```

This avoids having to use `Bifunctor` constraints in downstream instances, but at the cost of moving `bifunctors` further away from Haskell2010. (Granted, this PR is _also_ moving away from Haskell2010.)

This trick doesn't work for the `Functor` instance for `Biff`, which crucially relies on `p` being a `Bifunctor`. But perhaps this is a sign that there shouldn't be a `Profunctor` instance for `Biff` at all. After all, there is already a split between `profunctors`' `Cayley` and `bifunctors`' `Tannen` (which are otherwise identical), so perhaps there should be a `profunctors`-specific counterpart to `bifunctors`' `Biff`.

Reply:
If `Bifunctor` picks up the quantified superclass constraint like `Profunctor`, then there is no manufactured dilemma. Both of them can adopt the weaker `(Functor f, Functor (p a)) =>` context.

Reply:
I've changed `bifunctors` to match this plan in HEAD.